Specialize empty varbin const compare

AdamGS opened the pull request:
I'll take multiple double digit % improvements in clickbench.

AdamGS committed:
.

joseph-isaacs committed:
fix: min_max compute stat for chunked (#2228)

We now return early if there are no stats values in any chunk

joseph-isaacs commented on vortex-array/src/array/varbin/compute/compare.rs:
```diff
@@ -20,6 +24,40 @@ impl CompareFn<VarBinArray> for VarBinEncoding {
             let nullable = lhs.dtype().is_nullable() || rhs_const.dtype().is_nullable();
             let len = lhs.len();
 
+            if matches!(operator, Operator::Eq | Operator::NotEq) {
```
Can you do all operators.
```maths!
a < "" iff false
a =< "" iff a == ""
a > "" if a != ""
a >= "" true
```

joseph-isaacs committed:
chore: added a comment to explain chunked min max (#2229)

AdamGS committed:
Host our own Clickbench parquet files (#2230)

1. Will make us better neighbors, consuming less bandwidth from others.
2. Save us some time/cpu fixing the types on every run.

All files are hosted in a publicly readable R2 bucket, it might have
some rate limiting applied but if we run into that, fixing it should be
as easy as hosting it under our own custom domain.

AdamGS commented:
~@joseph-isaacs noted that we can also have a nice similar optimization for views, so I'm adding it now.~ nvm :(

AdamGS committed:
feat: Select filtered child in the browser (#2224)

fixes #2223.

AdamGS committed:
.

AdamGS committed:
Merge branch 'develop' into adamg/optimize-varbin-cmp-empty-const-rhs

AdamGS committed:
.

joseph-isaacs commented on vortex-array/src/array/varbin/compute/compare.rs:
```diff
@@ -20,6 +24,38 @@ impl CompareFn<VarBinArray> for VarBinEncoding {
+            if rhs_is_empty {
+                let lhs_offsets = lhs.offsets().into_canonical()?.into_primitive()?;
+
+                let buffer = match_each_native_ptype!(lhs_offsets.ptype(), |$P| {
```
can you make this a func, so we only have single line macros

AdamGS committed:
.

joseph-isaacs reviewed (approved):
LG, nice
